be/src/util/compression/block_compression.cpp

```diff
@@ -753,6 +767,21 @@ class ZstdBlockCompression final : public BlockCompressionCodec {
         }
         compression::ZSTDCompressionContext* context = ref.value().get();
         ZSTD_CCtx* ctx = context->ctx;
+        size_t ret;
+
+        // Every zstd compression context get from pool will be inited by default
+        // with level = ZSTD_CLEVEL_DEFAULT(3). And the context will be return to the
+        // pool by reseting back to level = ZSTD_CLEVEL_DEFAULT(3).
+        // What we should do here is simply set the level as we wanted.
+        if (_level != -1) {
+            DCHECK(_level >= 1 && _level <= 22);
```

Comment:
DHCECK is not good enough. In release mode it is nothing. shall fail and error out if the level is beyond [1, 22],

Reply:
Fixed